pkg/cmd/target.go

```diff
@@ -741,3 +760,26 @@ func shootWrapper(targetReader TargetReader, targetWriter TargetWriter, configRe
 	}
 	return nil
 }
+
+//set namespace for current kubectl ctx
+func setKubectlNamespace(kubectlNameSpace string) error {
+	if kubectlNameSpace == "" {
+		return errors.New("Namespace must be provided")
+	}
+
+	out1, err1 := exec.Command("bash", "-c", "gardenctl kubectl config current-context").Output()
```

Comment:
Use `ExecCmdReturnOutput`  with the the kubeconfig instead of calling the gardenctl binary.